src/Tribe/Venue.php

```diff
@@ -420,50 +430,76 @@ public function has_venue_data( $data ) {
 	 * Updates an venue
 	 *
 	 * @param int   $venue_id The venue ID to update.
-	 * @param array $data    The venue data.
+	 * @param array $data     The venue data.
 	 *
+	 * @return int The updated venue post ID
 	 */
 	public function update( $venue_id, $data ) {
-		$data['ShowMap']     = isset( $data['ShowMap'] ) ? $data['ShowMap'] : 'false';
-		$data['ShowMapLink'] = isset( $data['ShowMapLink'] ) ? $data['ShowMapLink'] : 'false';
-
-		$args = array(
-			'ID' => $venue_id,
-		);
+		/**
+		 * Filters the ID of the venue before the class updates it.
+		 *
+		 * If a non `null` value is returned that will be returned and the venue update process will bail.
+		 *
+		 * @param mixed  $check       Whether the venue update process should proceed or not.
+		 * @param int    $venue_id    The post ID of the venue that should be updated
+		 * @param array  $data        The data provided to update the venue.
+		 *
+		 * @since TBD
+		 */
+		$check = apply_filters( 'tribe_events_tribe_venue_update', null, $venue_id, $data );
 
-		if ( isset( $data['post_title'] ) ) {
-			$args['post_title'] = $data['post_title'];
+		if ( null !== $check ) {
+			return $check;
 		}
 
-		if ( isset( $data['Venue'] ) ) {
-			$args['post_title'] = $data['Venue'];
-			unset( $data['Venue'] );
-		}
+		$data = new Tribe__Data( $data, '' );
 
-		if ( isset( $data['post_content'] ) ) {
-			$args['post_content'] = $data['post_content'];
-		}
+		unset( $data['VenueID'] );
 
-		if ( isset( $data['Description'] ) ) {
-			$args['post_content'] = $data['Description'];
-			unset( $data['Description'] );
-		}
+		$args = array_filter( array(
+			'ID'            => $venue_id,
+			'post_title'    => Tribe__Utils__Array::get( $data, 'post_title', $data['Venue'] ),
```

Comment:
the `isset($data['Venue'])` is done implicitly in the `Tribe__Data` object; the syntax you use in your example is the one used to get `$data['post_title']['Venue']` already.